OpenRA.Mods.Common/Activities/Air/Land.cs

```diff
@@ -42,22 +46,28 @@ public override Activity Tick(Actor self)
 					return this;
 			}
 
-			if (!target.IsValidFor(self))
+			// TargetType.Invalid indicates target was an actor and either killed, removed or teleported
+			if (target.Type == TargetType.Invalid)
+			{
+				Cancel(self);
 				return NextActivity;
+			}
 
 			if (IsCanceling)
 			{
 				aircraft.RemoveInfluence();
 				return NextActivity;
 			}
 
-			if (requireSpace && !landingInitiated)
+			if (!landingInitiated)
 			{
-				var landingCell = self.World.Map.CellContaining(target.CenterPosition);
-				if (!aircraft.CanLand(landingCell, ignoreActor))
+				var landingCell = !aircraft.Info.VTOL ? self.World.Map.CellContaining(target.CenterPosition + offset) : self.Location;
```

Comment:
Why `self.Location` for VTOL aircraft but `target.CenterPosition + offset` for everything else?

I realize that this question is coming close to a (i'm sure very annoying) scope creep: this activity assumes/requires that the actor is already at the correct X,Y position where it really should be checking this and queuing a fly child activity if it isn't.  That's out of scope for this PR, but until then I don't see why the VTOL distinction here actually matters.